src/Symfony/Component/Security/Guard/Provider/GuardAuthenticationProvider.php

```diff
@@ -0,0 +1,140 @@
+<?php
+
+namespace Symfony\Component\Security\Guard\Provider;
```

Comment:
I propose to move this namespace into `Guard\Authentication`, just as done in Security Core. Or move the class in `Guard\Authentication` directly (to avoid long FQCNs). The -AuthenticationProvider prefix already makes it clear it's an authentication provider, no need to duplicate that in the namespaces imo (in my ideal world, it would be `Guard\Authentication\Provider\Guard`).